runtime/common_template_instantiations.cpp

```diff
@@ -0,0 +1,32 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2022 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "runtime/common_template_instantiations.h"
+
+template class array<int64_t>;
+template class array<string>;
+template class array<double>;
+template class array<bool>;
+template class array<mixed>;
+
+template class array<Optional<int64_t>>;
+template class array<Optional<string>>;
+template class array<Optional<double>>;
+template class array<Optional<bool>>;
+
+template class array<array<int64_t>>;
+template class array<array<string>>;
+template class array<array<double>>;
+template class array<array<bool>>;
+template class array<array<mixed>>;
+
+template class Optional<int64_t>;
+template class Optional<string>;
+template class Optional<double>;
```

Comment:
Optional<bool> also, it represents "null"

Reply:
It's skipped intentionally because there is an explicit specialization for `Optional<bool>`. And in this case explicit instantiation has no effect, see [cppreference](https://en.cppreference.com/w/cpp/language/class_template#:~:text=Explicit%20instantiation%20has%20no%20effect%20if%20an%20explicit%20specialization%20appeared%20before%20for%20the%20same%20set%20of%20template%20arguments.)